RUFAS/biophysical/manure/separators/separator.py

```diff
@@ -0,0 +1,133 @@
+        The efficiency of the separator in removing ash from the manure.
+    non_degradable_volatile_solids_efficiency : float
+        The efficiency of the separator in removing non-degradable volatile solids from the manure.
+    degradable_volatile_solids_efficiency : float
+        The efficiency of the separator in removing degradable volatile solids from the manure.
+    total_solids_efficiency : float
+        The efficiency of the separator in removing total solids from the manure.
+    """
+
+    water_efficiency: float
+    ammoniacal_nitrogen_efficiency: float
+    nitrogen_efficiency: float
+    phosphorus_efficiency: float
+    potassium_efficiency: float
+    ash_efficiency: float
+    non_degradable_volatile_solids_efficiency: float
+    degradable_volatile_solids_efficiency: float
+    total_solids_efficiency: float
+
+
+class Separator(Processor):
+    """
+    A manure processor that separates solids from liquids.
+
+    Parameters
+    ----------
+    config : SeparatorConfig
+        Configuration for the separator.
+    is_housing_emissions_calculator : bool
+        Indicates if a Processor calculates housing emissions.
+
+    Attributes
+    ----------
+    config : SeparatorConfig
+        Configuration for the separator.
+    is_housing_emissions_calculator : bool
+        If true, processor will only accept ManureStreams with non-None PenManureData, if false then vice versa.
+
+    """
+
+    def __init__(self, config: SeparatorConfig, is_housing_emissions_calculator: bool = False) -> None:
+        """Initializes a new Separator."""
+        super().__init__(is_housing_emissions_calculator)
+        self.config: SeparatorConfig = config
+        self.held_manure: ManureStream | None = None
+
+    def receive_manure(self, manure: ManureStream) -> None:
+        """
+        Takes in manure to be processed.
+
+        Parameters
+        ----------
+        manure : ManureStream
+            The manure to be processed.
+
+        """
+        if self.held_manure is None:
+            self.held_manure = manure
+        else:
+            self.held_manure += manure
+
+    def process_manure(self, conditions: CurrentDayConditions, time: Time) -> dict[str, ManureStream]:
+        """
+        Executes the daily manure processing operations.
+
+        Parameters
+        ----------
+        conditions : CurrentDayConditions
+            Current weather and environmental conditions that manure is being processed in.
+        time : Time
+            Time instance containing the simulations temporal information.
+
+        Returns
+        -------
+        dict
+            A dictionary containing:
+            - "solid" : ManureStream
+                The solid portion of the separated manure.
+            - "liquid" : ManureStream
+                The liquid portion of the separated manure.
+
+        """
+        if not self.held_manure:
+            return {"solid": ManureStream(), "liquid": ManureStream()}
```

Comment:
```suggestion
            return {}
```
This isn't specified in the design doc (that's my bad), but I was thinking that if there is no manure to be passed on to the next Processor then it would be best to return an empty dict. I see two main benefits to this:
1. It is more explicit to someone looking at what is being passed between Processors that there is no mass being transferred, because they can see that the dict is empty instead of looking at the dict, then having to look at the attributes inside each `ManureStream`.
2. It saves a small amount of processing power, because the `ManureManager` will see that the dict is empty and just move on to the next Processor. Instead of passing empty `ManureStream`s on to the next Processor, which will still have to be added even though they do not have any mass.